src/app/features/login/login.component.ts

```diff
@@ -1,14 +1,42 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, EventEmitter, Output, ViewChild } from '@angular/core';
+
+import { APP_ROUTS, User } from 'src/app/app.model';
+import { faEye, faEyeSlash } from '@fortawesome/free-solid-svg-icons';
 
 @Component({
   selector: 'app-login',
   templateUrl: './login.component.html',
   styleUrls: ['./login.component.scss'],
 })
-export class LoginComponent implements OnInit {
-  constructor() {}
+export class LoginComponent {
+  @ViewChild('formLogin') formLogin: any;
+
+  hidePassword = true;
+  eyeIcon = faEye;
+  eyeSlashIcon = faEyeSlash;
+  //for development
+  model: User = {
+    name: 'Yura',
+    email: 'Petya@gmai.sdf',
+    password: 'adsdfasf',
+  };
+
+  @Output() navigateEvent = new EventEmitter<APP_ROUTS>();
+  @Output() loginEvent = new EventEmitter<User>();
+
+  login() {
+    if (this.formLogin.form.status === 'VALID') {
+      this.loginEvent.emit(this.model);
+      this.navigateEvent.emit(APP_ROUTS.COURSES);
+    }
+  }
+
+  navigateToRegistration(event: Event) {
+    event.preventDefault();
+    this.navigateEvent.emit(APP_ROUTS.REGISTRATION);
+  }
 
-  ngOnInit(): void {
-    console.log('LoginComponent');
+  togglePassword() {
```

Comment:
It is better to keep all the logic in a component or services, leaving only variables in the template, also, these kinds of methods are commonly called on....Click().

As example:

```
onPasswordIconClick(): void {
    this.isPasswordHide = !this.isPasswordHide;
    this.passwordIcon = this.isPasswordHide ? faEye : faEyeSlash;
    this.passwordTextType = this.isPasswordHide ? 'password' : 'text';
  }
```
  
  ```
where isPasswordHide: boolean = true;
  passwordIcon: IconDefinition = faEye;
  passwordTextType: string = 'password';
```
  
  are listed above, and on template you have clear 
  `[type]="passwordTextType"` and  `[icon]="passwordIcon"`

Reply:
good idea! i changed in login and registration